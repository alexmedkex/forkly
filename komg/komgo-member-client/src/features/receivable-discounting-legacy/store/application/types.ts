import { ImmutableMap, stringOrNull } from '../../../../utils/types'
import {
  IReceivablesDiscountingInfo,
  IReceivablesDiscountingBase,
  ICreateReceivableDiscountingApplicationResponse,
  IHistory,
  IReceivablesDiscounting
} from '@komgo/types'
import { List, Map } from 'immutable'
import { Action } from 'redux'

export enum ReceivableDiscountingApplicationActionType {
  // create
  CREATE_APPLICATION_REQUEST = '@@receivable-discounting/CREATE_APPLICATION_REQUEST',
  CREATE_APPLICATION_SUCCESS = '@@receivable-discounting/CREATE_APPLICATION_SUCCESS',
  CREATE_APPLICATION_FAILURE = '@@receivable-discounting/CREATE_APPLICATION_FAILURE',

  // update
  UPDATE_APPLICATION_REQUEST = '@@receivable-discounting/UPDATE_APPLICATION_REQUEST',
  UPDATE_APPLICATION_SUCCESS = '@@receivable-discounting/UPDATE_APPLICATION_SUCCESS',
  UPDATE_APPLICATION_FAILURE = '@@receivable-discounting/UPDATE_APPLICATION_FAILURE',

  SHARE_APPLICATION_REQUEST = '@@receivable-discounting/SHARE_APPLICATION_REQUEST',
  SHARE_APPLICATION_SUCCESS = '@@receivable-discounting/SHARE_APPLICATION_SUCCESS',
  SHARE_APPLICATION_FAILURE = '@@receivable-discounting/SHARE_APPLICATION_FAILURE',

  // read
  FETCH_APPLICATION_REQUEST = '@@receivable-discounting/FETCH_APPLICATION_REQUEST',
  FETCH_APPLICATION_SUCCESS = '@@receivable-discounting/FETCH_APPLICATION_SUCCESS',
  FETCH_APPLICATION_FAILURE = '@@receivable-discounting/FETCH_APPLICATION_FAILURE',

  FETCH_MULTIPLE_APPLICATION_REQUEST = '@@receivable-discounting/FETCH_MULTIPLE_APPLICATION_REQUEST',
  FETCH_MULTIPLE_APPLICATION_SUCCESS = '@@receivable-discounting/FETCH_MULTIPLE_APPLICATION_SUCCESS',
  FETCH_MULTIPLE_APPLICATION_FAILURE = '@@receivable-discounting/FETCH_MULTIPLE_APPLICATION_FAILURE',

  FETCH_APPLICATION_HISTORY_REQUEST = '@@receivable-discounting/FETCH_APPLICATION_HISTORY_REQUEST',
  FETCH_APPLICATION_HISTORY_SUCCESS = '@@receivable-discounting/FETCH_APPLICATION_HISTORY_SUCCESS',
  FETCH_APPLICATION_HISTORY_FAILURE = '@@receivable-discounting/FETCH_APPLICATION_HISTORY_FAILURE'
}

export interface ApplicationError extends Action {
  type:
    | ReceivableDiscountingApplicationActionType.CREATE_APPLICATION_FAILURE
    | ReceivableDiscountingApplicationActionType.FETCH_MULTIPLE_APPLICATION_FAILURE
  payload: stringOrNull
}

export interface CreateSucceeded extends Action {
  type: ReceivableDiscountingApplicationActionType.CREATE_APPLICATION_SUCCESS
  payload: ICreateReceivableDiscountingApplicationResponse
}

export interface FetchMultipleSucceeded extends Action {
  type: ReceivableDiscountingApplicationActionType.FETCH_MULTIPLE_APPLICATION_SUCCESS
  payload: IReceivablesDiscountingInfo[]
}

export interface FetchSucceeded extends Action {
  type: ReceivableDiscountingApplicationActionType.FETCH_APPLICATION_SUCCESS
  payload: IReceivablesDiscountingInfo
}

export interface UpdateSucceeded extends Action {
  type: ReceivableDiscountingApplicationActionType.UPDATE_APPLICATION_SUCCESS
  payload: IReceivablesDiscountingBase
  rdId: string
}

export interface FetchHistorySucceeded extends Action {
  type: ReceivableDiscountingApplicationActionType.FETCH_APPLICATION_HISTORY_SUCCESS
  payload: IHistory<IReceivablesDiscounting>
  rdId: string
}

export interface ReceivableDiscountingApplicationStateProperties {
  byId: Map<string, IReceivablesDiscountingInfo>
  historyById: Map<string, IHistory<IReceivablesDiscounting>>
  ids: List<string>
  error: stringOrNull
}

export type ReceivableDiscountingApplicationState = ImmutableMap<ReceivableDiscountingApplicationStateProperties>

export type ReceivableDiscountingApplicationAction =
  | CreateSucceeded
  | FetchHistorySucceeded
  | FetchMultipleSucceeded
  | FetchSucceeded
  | UpdateSucceeded
  | ApplicationError
