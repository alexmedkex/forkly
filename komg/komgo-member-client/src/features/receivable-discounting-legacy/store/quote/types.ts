import { IQuoteBase, IQuote, IHistory } from '@komgo/types'
import { Action } from 'redux'
import { stringOrNull, ImmutableMap } from '../../../../utils/types'
import { Map } from 'immutable'

export enum QuoteActionType {
  // create
  SHARE_QUOTE_REQUEST = '@@receivable-discounting/SHARE_QUOTE_REQUEST',
  SHARE_QUOTE_SUCCESS = '@@receivable-discounting/SHARE_QUOTE_SUCCESS',
  SHARE_QUOTE_FAILURE = '@@receivable-discounting/SHARE_QUOTE_FAILURE',

  // update
  UPDATE_QUOTE_REQUEST = '@@receivable-discounting/UPDATE_QUOTE_REQUEST',
  UPDATE_QUOTE_SUCCESS = '@@receivable-discounting/UPDATE_QUOTE_SUCCESS',
  UPDATE_QUOTE_FAILURE = '@@receivable-discounting/UPDATE_QUOTE_FAILURE',

  // read
  FETCH_QUOTE_REQUEST = 'FETCH_QUOTE_REQUEST',
  FETCH_QUOTE_FAILURE = 'FETCH_QUOTE_FAILURE',
  FETCH_QUOTE_SUCCESS = 'FETCH_QUOTE_SUCCESS',

  FETCH_QUOTE_HISTORY_REQUEST = 'FETCH_QUOTE_HISTORY_REQUEST',
  FETCH_QUOTE_HISTORY_SUCCESS = 'FETCH_QUOTE_HISTORY_SUCCESS',
  FETCH_QUOTE_HISTORY_FAILURE = 'FETCH_QUOTE_HISTORY_FAILURE'
}

export interface UpdateSucceeded extends Action {
  type: QuoteActionType.UPDATE_QUOTE_SUCCESS
  payload: IQuoteBase
}

export interface QuoteStateProperties {
  byId: Map<string, IQuote>
  historyById: Map<string, IHistory<IQuote>>
  error: stringOrNull
}

export interface FetchHistorySucceeded extends Action {
  type: QuoteActionType.FETCH_QUOTE_HISTORY_SUCCESS
  payload: IHistory<IQuote>
  quoteId: string
}

export type QuoteState = ImmutableMap<QuoteStateProperties>

export type QuoteAction = UpdateSucceeded | FetchHistorySucceeded
