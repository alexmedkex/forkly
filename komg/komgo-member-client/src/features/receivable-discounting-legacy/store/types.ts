import {
  IParticipantRFPSummary,
  IQuoteBase,
  IRFPSummariesResponse,
  ParticipantRFPStatus,
  ICargo,
  ITrade,
  RDStatus,
  IHistory
} from '@komgo/types'
import { Map } from 'immutable'
import { Action } from 'redux'
import { ImmutableMap, stringOrNull } from '../../../utils/types'
import { Counterparty } from '../../counterparties/store/types'
import { DocumentContextBase } from '../../trades/store/types'

// Only use REQUEST/SUCCESS/FAILURE at the end of action type names
export enum ReceivableDiscountingActionType {
  CREATE_REQUEST_FOR_PROPOSAL_REQUEST = '@@receivable-discounting/CREATE_RFP_REQUEST',
  CREATE_REQUEST_FOR_PROPOSAL_SUCCESS = '@@receivable-discounting/CREATE_RFP_SUCCESS',
  CREATE_REQUEST_FOR_PROPOSAL_FAILURE = '@@receivable-discounting/CREATE_RFP_FAILURE',

  FETCH_RFP_SUMMARY_REQUEST = '@@receivable-discounting/FETCH_RFP_SUMMARY_REQUEST',
  FETCH_RFP_SUMMARY_SUCCESS = '@@receivable-discounting/FETCH_RFP_SUMMARY_SUCCESS',
  FETCH_RFP_SUMMARY_FAILURE = '@@receivable-discounting/FETCH_RFP_SUMMARY_FAILURE',

  CREATE_QUOTE_REQUEST = '@@receivable-discounting/CREATE_QUOTE_REQUEST',
  CREATE_QUOTE_SUCCESS = '@@receivable-discounting/CREATE_QUOTE_SUCCESS',
  CREATE_QUOTE_FAILURE = '@@receivable-discounting/CREATE_QUOTE_FAILURE',

  FETCH_PARTICIPANT_RFP_SUMMARY_REQUEST = '@@receivable-discounting/FETCH_PARTICIPANT_RFP_SUMMARY_REQUEST',
  FETCH_PARTICIPANT_RFP_SUMMARY_SUCCESS = '@@receivable-discounting/FETCH_PARTICIPANT_RFP_SUMMARY_SUCCESS',
  FETCH_PARTICIPANT_RFP_SUMMARY_FAILURE = '@@receivable-discounting/FETCH_PARTICIPANT_RFP_SUMMARY_FAILURE',

  FETCH_QUOTE_REQUEST = '@@receivable-discounting/FETCH_QUOTE_REQUEST',
  FETCH_QUOTE_SUCCESS = '@@receivable-discounting/FETCH_QUOTE_SUCCESS',
  FETCH_QUOTE_FAILURE = '@@receivable-discounting/FETCH_QUOTE_FAILURE',

  SUBMIT_QUOTE_REQUEST = '@@receivable-discounting/SUBMIT_QUOTE_REQUEST',
  SUBMIT_QUOTE_SUCCESS = '@@receivable-discounting/SUBMIT_QUOTE_SUCCESS',
  SUBMIT_QUOTE_FAILURE = '@@receivable-discounting/SUBMIT_QUOTE_FAILURE',

  REJECT_RFP_REQUEST = '@@receivable-discounting/REJECT_RFP_REQUEST',
  REJECT_RFP_SUCCESS = '@@receivable-discounting/REJECT_RFP_SUCCESS',
  REJECT_RFP_FAILURE = '@@receivable-discounting/REJECT_RFP_FAILURE',

  ACCEPT_QUOTE_REQUEST = '@@receivable-discounting/ACCEPT_QUOTE_REQUEST',
  ACCEPT_QUOTE_SUCCESS = '@@receivable-discounting/ACCEPT_QUOTE_SUCCESS',
  ACCEPT_QUOTE_FAILURE = '@@receivable-discounting/ACCEPT_QUOTE_FAILURE',

  FETCH_TRADE_HISTORY_REQUEST = '@@receivable-discounting/FETCH_TRADE_HISTORY_REQUEST',
  FETCH_TRADE_HISTORY_SUCCESS = '@@receivable-discounting/FETCH_TRADE_HISTORY_SUCCESS',
  FETCH_TRADE_HISTORY_FAILURE = '@@receivable-discounting/FETCH_TRADE_HISTORY_FAILURE'
}

export interface IMemberMarketSelectionItem {
  counterparty: Counterparty
  location: string
  appetite: string
  availability: string
  creditLimit: string
  riskFee: string
  margin: string
  maxTenor: string
}

export interface IReceivableDiscountingDashboardTrader {
  tradeTechnicalId: string
  rdId?: string
  tradeDate: string | number | Date
  tradeId: string
  counterparty: string
  bank: string
  commodity: string
  invoiceAmount?: string
  invoiceType?: string
  currency: string
  status: string
  rdStatus?: RDStatus
}

export interface IReceivableDiscountingDashboardBank {
  tradeId: string
  seller: string
  buyer: string
  paymentTerms: string
  rd: {
    status: RDStatus
    staticId: string
  }
  invoiceAmount: string
  invoiceType: string
  currency: string
  discountingDate: string | number | Date
  discountingDateType: string
  status: string
  requestDate: string
}

export interface IRFPRequestSummary {
  companyStaticId: string
  status: ParticipantRFPStatus
}

export interface ISubmitQuoteFormDetails extends IQuoteBase {
  comment?: string
  rdId: string
}

// Move to @komgo/types
export interface IRFPReply {
  rdId: string
  comment?: string
}

export interface IBankDeclineRFPFormDetails {
  comment?: string
}

export interface IFetchMultipleReceivableDiscountFilter {
  tradeSourceIds: string[]
}

export interface ICreateRequestForProposal {
  rdId: string
  participantStaticIds: string[]
}

export interface ITradeSnapshot {
  source: string
  sourceId: string
  trade: ITrade
  movements: ICargo[]
  createdAt?: string
  updatedAt?: string
}

// Move to @komgo/types
export interface IQuoteSubmission extends IRFPReply {
  quoteId: string
}

// Move to @komgo/types
export interface IQuoteAcceptSubmission extends IQuoteSubmission {
  participantStaticId: string
}

export interface RDDocumentContext extends DocumentContextBase {
  rdId: string
}

export interface CreateRequestForProposalError extends Action {
  type: ReceivableDiscountingActionType.CREATE_REQUEST_FOR_PROPOSAL_FAILURE
  payload: stringOrNull
}

export interface ReceivableDiscountingFetchedQuotesAction extends Action {
  type: ReceivableDiscountingActionType.FETCH_RFP_SUMMARY_SUCCESS
  payload: IRFPSummariesResponse
  rdId: string
}

export interface ReceivableDiscountingFetchedTradeHistoryAction extends Action {
  type: ReceivableDiscountingActionType.FETCH_TRADE_HISTORY_SUCCESS
  payload: IHistory<ITradeSnapshot>
  sourceId: string
}

export interface ReceivableDiscountingStateProperties {
  rfpSummariesByRdId: Map<string, IParticipantRFPSummary[]>
  rfpSummariesByRdIdByParticipantStaticId: Map<string, Map<string, IParticipantRFPSummary>>
  tradeSnapshotHistoryById: Map<string, IHistory<ITradeSnapshot>>
  error: stringOrNull
}

export type ReceivableDiscountingState = ImmutableMap<ReceivableDiscountingStateProperties>

export type ReceivableDiscountingAction =
  | ReceivableDiscountingFetchedQuotesAction
  | CreateRequestForProposalError
  | ReceivableDiscountingFetchedTradeHistoryAction
