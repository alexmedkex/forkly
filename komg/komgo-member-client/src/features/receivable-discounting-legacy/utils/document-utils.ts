import { Document } from '../../document-management'
import { isLaterThan } from '../../../utils/date'

export type ModifiedDocument = Document & { lastModifiedAt: Date }

const latestModified = (doc1?: ModifiedDocument, doc2?: ModifiedDocument): ModifiedDocument => {
  if (doc1 && doc2) {
    return isLaterThan(doc1.lastModifiedAt, doc2.lastModifiedAt) ? doc1 : doc2
  }
  return doc1 ? doc1 : doc2
}

export const latestDocument = (docs: Document[]): ModifiedDocument | undefined =>
  docs
    .map(doc => ({
      ...doc,
      lastModifiedAt: doc.receivedDate || doc.registrationDate
    }))
    .reduce(latestModified, undefined as any)
