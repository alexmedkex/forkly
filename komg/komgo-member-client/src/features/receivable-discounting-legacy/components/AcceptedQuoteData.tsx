import { IQuote, IQuoteBase, IHistory, RequestType, DiscountingType, IReceivablesDiscounting } from '@komgo/types'
import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import { MinimalAccordionWrapper } from '../../../components/accordion/MinimalAccordionWrapper'
import { DisplayQuote } from '../../receivable-finance/entities/quote/components/DisplayQuote'
import { Button } from 'semantic-ui-react'
import { Formik, FormikProps } from 'formik'
import { EditAcceptedQuoteFields } from '../../receivable-finance/pages/view-request/components/EditAcceptedQuoteFields'
import { MultiErrorMessage } from '../../../components/error-message'
import { formikQuoteAcceptedErrors } from '../utils/errors'
import { decorateQuoteForInitialValues, createEditQuoteValidator } from '../utils/edit-utils'
import { StatusText } from './generics/StatusText'

export interface IAcceptedQuoteData {
  index: string
  open: boolean
  quote: IQuote
  isSubmitting: boolean
  isEditing: boolean
  editable: boolean
  comment?: string
  replyDate?: string
  provider?: string
  agreedTermsHistory: IHistory<IQuote>
  changed: boolean
  rd: IReceivablesDiscounting
  handleSubmit: (values: IQuoteBase) => void
  handleEditClicked: () => void
  handleCancelClicked: () => void
  handleToggleAccordion: () => void
}

const AcceptedQuoteData: React.FC<IAcceptedQuoteData> = ({
  index,
  open,
  quote,
  comment,
  replyDate,
  provider,
  editable,
  isEditing,
  changed,
  isSubmitting,
  handleCancelClicked,
  handleEditClicked,
  handleToggleAccordion,
  handleSubmit,
  agreedTermsHistory,
  rd
}) => {
  const initialValues = quote ? decorateQuoteForInitialValues(quote) : {}
  const actions = formik =>
    isEditing ? (
      <>
        <Button
          content={'Save'}
          data-test-id="save-accepted-quote"
          primary={true}
          type="submit"
          disabled={isSubmitting}
          onClick={e => {
            e.stopPropagation()
            formik.handleSubmit(e)
          }}
        />
        <Button
          content={'Cancel'}
          data-test-id="savecancel-edit-accepted-quote"
          disabled={isSubmitting}
          onClick={e => {
            e.stopPropagation()
            formik.resetForm()
            handleCancelClicked()
          }}
        />
      </>
    ) : editable ? (
      <Button
        content={'Edit'}
        data-test-id="edit-accepted-quote"
        onClick={e => {
          e.stopPropagation()
          handleEditClicked()
        }}
      />
    ) : null

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={handleSubmit}
      validate={createEditQuoteValidator(initialValues, rd)}
      validateOnBlur={false}
      validateOnChange={true}
      render={(formik: FormikProps<IQuoteBase>) => (
        <>
          <MinimalAccordionWrapper
            buttons={actions(formik)}
            active={open}
            handleClick={handleToggleAccordion}
            index={index}
            highlight={changed}
            title="Agreed terms"
          >
            {formikQuoteAcceptedErrors(formik).length > 0 ? (
              <MultiErrorMessage
                data-test-id="edit-quote-accepted-validation-errors"
                title=""
                messages={formikQuoteAcceptedErrors(formik)}
              />
            ) : null}
            <StatusText data-test-id="quote-section-status-text" margin="0 0 0 28px">
              <UpperCaseText>Quote accepted - {moment(quote.createdAt).format('DD/MM/YYYY [at] HH:mm')}</UpperCaseText>
            </StatusText>
            {isEditing ? (
              <EditAcceptedQuoteFields data-test-id="edit-accepted-quote-panel" formik={formik} rd={rd} />
            ) : (
              <DisplayQuote
                quote={quote}
                provider={provider}
                comment={comment}
                replyDate={replyDate}
                agreedTermsHistory={agreedTermsHistory}
                sectionName="agreedTerms"
                requestType={rd.requestType}
                discountingType={rd.discountingType}
              />
            )}
          </MinimalAccordionWrapper>
        </>
      )}
    />
  )
}

const UpperCaseText = styled.span`
  &&& {
    text-transform: uppercase;
  }
`

export default AcceptedQuoteData
