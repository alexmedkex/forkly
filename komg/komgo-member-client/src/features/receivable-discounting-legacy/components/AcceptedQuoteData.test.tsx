import AcceptedQuoteData, { IAcceptedQuoteData } from './AcceptedQuoteData'
import { buildFakeQuote, RequestType, DiscountingType, buildFakeReceivablesDiscountingExtended } from '@komgo/types'
import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme'
import { Formik } from 'formik'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { ReceivableDiscountingViewPanels } from '../utils/constants'
import moment from 'moment'

describe('AcceptedQuoteData', () => {
  const requestType = RequestType.Discount
  const discountingType = DiscountingType.WithoutRecourse
  let testProps: IAcceptedQuoteData

  beforeEach(() => {
    moment.tz.guess = jest.fn(() => 'UTC')
    moment.tz.setDefault('UTC')
    Date.now = jest.fn(() => 1487076708000)

    testProps = {
      quote: buildFakeQuote({}, false, requestType, discountingType),
      index: ReceivableDiscountingViewPanels.AcceptedQuote,
      open: true,
      editable: true,
      isSubmitting: false,
      isEditing: false,
      changed: false,
      agreedTermsHistory: {},
      handleSubmit: jest.fn(),
      handleEditClicked: jest.fn(),
      handleCancelClicked: jest.fn(),
      handleToggleAccordion: jest.fn(),
      rd: buildFakeReceivablesDiscountingExtended(false, {
        requestType: RequestType.Discount,
        discountingType: DiscountingType.WithoutRecourse
      })
    }
  })

  it('renders correctly', () => {
    expect(renderer.create(<AcceptedQuoteData {...testProps} />).toJSON()).toMatchSnapshot()
  })

  it('should show an edit button if editable', () => {
    const component = mount(<AcceptedQuoteData {...testProps} />)

    expect(
      component
        .find('[data-test-id="edit-accepted-quote"]')
        .first()
        .exists()
    ).toBeTruthy()
  })

  it('should not show an edit button if not editable', () => {
    testProps.editable = false
    const component = mount(<AcceptedQuoteData {...testProps} />)

    expect(
      component
        .find('[data-test-id="edit-accepted-quote"]')
        .first()
        .exists()
    ).toBeFalsy()
  })

  describe('Edit', () => {
    let component: ReactWrapper
    beforeEach(() => {
      testProps.isEditing = true
      component = mount(<AcceptedQuoteData {...testProps} />)
    })

    it('should show an editable panel but not the info panel if editing', () => {
      testProps.isEditing = true

      component = mount(<AcceptedQuoteData {...testProps} />)

      expect(component.find('[data-test-id="edit-accepted-quote-panel"]').exists()).toBeTruthy()
    })

    it('should show info if not editing', () => {
      testProps.isEditing = false

      component = mount(<AcceptedQuoteData {...testProps} />)

      expect(component.find('[data-test-id="edit-accepted-quote-panel"]').exists()).toBeFalsy()
      expect(component.find('[data-test-id="agreedTerms-field-component-advanceRate"]').exists()).toBeTruthy()
    })

    it('should submit edits if a valid change is made', async done => {
      textInput(component, 'advanceRate', 50)

      component
        .find('[data-test-id="save-accepted-quote"]')
        .first()
        .simulate('click')

      setTimeout(() => {
        expect(testProps.handleSubmit).toHaveBeenCalled()
        done()
      }, 50)
    })

    it('should not submit edits if nothing is changed', async done => {
      component
        .find('[data-test-id="save-accepted-quote"]')
        .first()
        .simulate('click')

      setTimeout(() => {
        expect(testProps.handleSubmit).not.toHaveBeenCalled()
        done()
      }, 0)
    })

    it('should not submit edits if an invalid change is made', async done => {
      textInput(component, 'advanceRate', -1)

      component
        .find('[data-test-id="save-accepted-quote"]')
        .first()
        .simulate('click')

      setTimeout(() => {
        expect(testProps.handleSubmit).not.toHaveBeenCalled()
        done()
      }, 0)
    })

    it('should show errors if error is present and editing', () => {
      const Children = component.find(Formik).prop('render') as any

      const childrenWrapper = shallow(<Children errors={{ advanceRate: 'invalid' }} touched={{ advanceRate: true }} />)

      const errors = childrenWrapper.find('[data-test-id="edit-quote-accepted-validation-errors"]')
      expect(errors.exists()).toBeTruthy()
      expect(errors.prop('messages')).toEqual(['invalid'])
    })
  })

  const textInput = (component: ShallowWrapper | ReactWrapper, name: string, value: string | number) => {
    const input = component.find(`input[name="${name}"]`).first()
    input.simulate('focus')
    input.simulate('change', { target: { value, name } })
    input.simulate('blur')
  }
})
