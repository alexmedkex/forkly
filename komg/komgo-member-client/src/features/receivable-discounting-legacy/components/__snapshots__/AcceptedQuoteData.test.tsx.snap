// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptedQuoteData renders correctly 1`] = `
.c1.c1.c1 {
  color: #5d768f;
  font-size: 11px;
  font-weight: 600;
  line-height: 21px;
  width: auto;
  text-align: left;
  margin: 0 0 0 28px;
}

.c0.c0.c0 .komgo-accordion-header {
  height: 53px;
  padding: 15px;
  background-color: #ffffff;
  border: 1px solid #c0cfde;
  border-radius: 4px;
  box-shadow: 0 1px 4px 0 rgba(192,207,222,0.5);
  color: #1c2936;
  font-size: 14px;
  font-weight: 600;
  line-height: 21px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.c0.c0 .accordion-title-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 40%;
}

.c0.c0.c0 .accordion-title {
  margin-top: 1px;
}

.c0.c0.c0 .accordion-icon {
  color: #62768d;
  margin: -8px 5px 0px 5px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.c0.c0.c0 .accordion-status-wrapper {
  margin-top: 1px;
}

.c0.c0.c0 .accordion-button-wrapper {
  margin-top: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  min-width: 20%;
}

.c0.c0.c0 .disabled {
  pointer-events: none;
}

.c0.c0.c0 .lively-divider {
  display: none;
  position: relative;
  top: 14px;
  border-bottom: 1px solid #dbe5ec;
  text-align: center;
  left: 50%;
  width: 0%;
  -webkit-animation: open-line 0.4s forwards;
  animation: open-line 0.4s forwards;
}

.c0.c0.c0 .active .komgo-accordion-header {
  border-radius: 4px 4px 0px 0px;
  border-bottom: 0;
}

.c0.c0.c0 .active .accordion-title {
  margin-top: 0px;
}

.c0.c0.c0 .active .accordion-status-wrapper {
  margin-top: 0px;
}

.c0.c0.c0 .active .accordion-button-wrapper {
  margin-top: 0px;
}

.c0.c0.c0 .active .komgo-accordion-content {
  border-radius: 0px 0px 4px 4px;
  border-top: none;
  border-color: #c0cfde;
  margin: -15px 0px 8px 0px;
}

.c0.c0.c0 .active .accordion-icon {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  margin: 0px 5px 0px 5px;
}

.c0.c0.c0 .active .lively-divider {
  display: block;
  left: 1rem;
  width: calc(100% - 2rem);
  top: -15px;
}

.c3 {
  list-style: none;
  margin: 0;
  padding: 10px 0 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  margin: 0;
  line-height: 2;
  white-space: nowrap;
  margin-bottom: 5px;
  max-width: 720px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c7 {
  white-space: normal;
  margin-left: 20px;
  max-width: calc(100% - 220px);
}

.c4 {
  display: block;
  margin-bottom: 10px;
}

.c4 p {
  vertical-align: top;
  display: inline-block;
}

.c6 {
  font-weight: bold;
  width: 220px;
  text-align: right;
}

.c2.c2.c2 {
  text-transform: uppercase;
}

@media (min-width:68.75em) {
  .c3 {
    padding-left: 40px;
  }
}

<div
  bordercolor="#c0cfde"
  className="accordion ui fluid c0"
>
  <div
    className="active title"
    data-test-id="agreed-terms-accordion-title"
    onClick={[Function]}
  >
    <div
      className="ui block header komgo-accordion-header"
    >
      <div
        className="accordion-title-wrapper"
      >
        <i
          aria-hidden="true"
          className="chevron right icon accordion-icon"
        />
        <div
          className="accordion-title"
          style={
            Object {
              "marginLeft": "0px",
            }
          }
        >
          Agreed terms
        </div>
      </div>
      <div
        className="accordion-status-wrapper"
      />
      <div
        className="accordion-button-wrapper"
      >
        <button
          className="ui button"
          data-test-id="edit-accepted-quote"
          onClick={[Function]}
          role="button"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
  <div
    className="content active"
    data-test-id="agreed-terms-accordion-content"
  >
    <div
      className="lively-divider"
    />
    <div
      className="ui segment komgo-accordion-content"
    >
      <p
        className="c1"
        data-test-id="quote-section-status-text"
      >
        <span
          className="c2"
        >
          Quote accepted - 
          31/01/2019 at 00:00
        </span>
      </p>
      <ul
        className="c3"
      >
        <li
          className="c4 c5"
          data-test-id="agreedTerms-field-component-advanceRate"
        >
          <p
            className="c6"
          >
            Advance rate
          </p>
          <p
            className="c7"
            data-test-id="agreedTerms-value-advanceRate"
          >
            10.00%
          </p>
        </li>
        <li
          className="c4 c5"
          data-test-id="agreedTerms-field-component-pricingType"
        >
          <p
            className="c6"
          >
            Pricing type
          </p>
          <p
            className="c7"
            data-test-id="agreedTerms-value-pricingType"
          >
            All in
          </p>
        </li>
        <li
          className="c4 c5"
          data-test-id="agreedTerms-field-component-pricingAllIn"
        >
          <p
            className="c6"
          >
            All in
          </p>
          <p
            className="c7"
            data-test-id="agreedTerms-value-pricingAllIn"
          >
            0.50%
          </p>
        </li>
        <li
          className="c4 c5"
          data-test-id="agreedTerms-field-component-numberOfDaysDiscounting"
        >
          <p
            className="c6"
          >
            Days of discounting
          </p>
          <p
            className="c7"
            data-test-id="agreedTerms-value-numberOfDaysDiscounting"
          >
            80 days
          </p>
        </li>
        <li
          className="c4 c5"
          data-test-id="agreedTerms-field-component-interestType"
        >
          <p
            className="c6"
          >
            Interest type
          </p>
          <p
            className="c7"
            data-test-id="agreedTerms-value-interestType"
          >
            Cost of funds
          </p>
        </li>
        <li
          className="c4 c5"
          data-test-id="agreedTerms-field-component-indicativeCof"
        >
          <p
            className="c6"
          >
            Indicative Cost of Funds
          </p>
          <p
            className="c7"
            data-test-id="agreedTerms-value-indicativeCof"
          >
            5.50%
          </p>
        </li>
        <li
          className="c4 c5"
          data-test-id="agreedTerms-field-component-feeCalculationType"
        >
          <p
            className="c6"
          >
            Fee calculation based on
          </p>
          <p
            className="c7"
            data-test-id="agreedTerms-value-feeCalculationType"
          >
            Straight discount
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
